### DISCUSSION TITLE
[BUG] can't use Site.Metadata.CSV sample

### DISCUSSION THREAD

**Discussion by watana2** [pnp/powershell#4574] [General]
Connect-PnPOnline -Url $SPOAdminURL -Interactive -ErrorAction Stop

There was a change in September this year to use app registration, but the script hasn't been updated to reflect the change.

**Comment by jackpoz** [comment]
Hi, could you please include a link to the script you are referring to ? Is it from the PnP Script Samples repository ?

**Comment by gautamdsheth** [comment]
@jackpoz  - its this one - https://github.com/pnp/powershell/tree/dev/samples/Site.Metadata.CSV 

Not sure if we should update it, the whole folder contains some samples, which are quite old. Maybe we move them to script samples ?

**Comment by jackpoz** [comment]
Oh, I never noticed there were samples there. Script samples repo sounds like the best place to store them 😃

**Comment by wilecoyotegenius** [comment]
@watana2 you may still use the script as it is, if you set the app registration client id as environment variable. Please look [here](https://pnp.github.io/powershell/articles/defaultclientid.html#by-setting-an-environment-variable)

**Comment by watana2** [comment]
@wilecoyotegenius Thank you. The next issue is that it outputs a zero-size SPOSiteURLs.csv file. I wonder if I missed something or if anyone has successfully run the script recently.
ResultRows                    : {}
RowCount                     : 0
TotalRows                      : 0

**Comment by gautamdsheth** [comment]
hi @watana2 , moving this to a discussion. We wont be working on updating/fixing them due to lack of time and resources. The community can help you from there.

We will be migrating some of these samples to https://pnp.github.io/script-samples/ in the coming weeks.

About your issue , I would double check the SiteID values or maybe update the query filter used in the `Submit-PnPSearchQuery` command.
